### PR TITLE
Added always_private argument to url() method

### PR DESCRIPTION
**Pull request by MasterGroosha**
# Description

There are some cases when you want to be sure that your link will work regardless of chat username. This change makes it so you can force creation of private link

## Type of change

Please delete options that are not relevant.

- [ ] Documentation (typos, code examples or any documentation update)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Not tested

# Checklist:

Lol no, I just edited on github

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by MasterGroosha** [comment]
Instead of "always_private" it can be named as "force_private". Maybe it will fit better

File: aiogram/types/message.py [hunk 1]
```diff
@@ -228,22 +228,23 @@ def html_text(self) -> str:
         return self.parse_entities()
 
     @property
-    def url(self) -> str:
+    def url(self, always_private=False) -> str:
```
Comment (uwinx) [inline aiogram/types/message.py:231]: @MasterGroosha, `url` is a `property`, so having any parameter except `self` in its getter is impossible. descriptor's getter has to have the following signature `(self, instance, owner) -> T` :)

You may take a look at https://docs.python.org/3/howto/descriptor.html and remember descriptors.

Comment (evgfilim1) [reply aiogram/types/message.py:231]: Oh, haven't noticed. :eyes:

**Comment by MasterGroosha** [comment]
@uwinx aiogram code is too overkill for me in terms of number of places you need to edit to add a damn single method, feel free to do that yourself.

**Comment by evgfilim1** [comment]
> number of places you need to edit to add a damn single method

@MasterGroosha For this particular feature you can just add a new property (as suggested by @uwinx) and slightly modify existing one to avoid code duplication. Everything can be done in the only one file. :shrug:

**Comment by uwinx** [comment]
@MasterGroosha, I'm very disappointed to see you given up and this PR closed. We appreciate your help. However, we can't approve changes that simply won't work. In our case I have suggested a solution, which would have brought exactly one new property and described why wouldn't it work as we want.

Thanks, reopen or comment on anytime as you wish.

**Comment by MasterGroosha** [comment]
Unfortunately, in my opinion this doesn’t look like a good solution. A better solution would be removing `url` property and adding a new method, for example, `get_url()`, which will be used as a unified way to get message URL.

However, this would be not backwards compatible as it will break the existing code. Also it would require writing tests which I’m not good with.

TL;DR: just adding a new property will lead to code duplication, potentially harming the framework in long run.
